content/rationale.md

```diff
@@ -0,0 +1,85 @@
+title: Why complate?
+
+Why complate?
+=============
+
+## Abstraction
+
+When we are developing our frontend components, we can use complate markup
+abstractions ("macros") as an abstraction layer for our component. Consider
+the following macro:
+
+```jsx
+function Navbar (_props, ...children) {
+  return <nav>
+    {children}
+  </nav>
+}
+```
```

Comment:
That example strikes me as a little too minimalistic, as it omits parameters (which also results in that somewhat alienating, unexplained `_props` argument - which, btw, I intentionally call "param[eter]s").

Here's what I tend to use:

```jsx
<Card title="Hello World">
    <p>lorem ipsum dolor sit amet</p>
    <Button label="toggle" />
</Card>
```

(though I'd probably omit `Button` for starters)

Also, in the interest of consistency: I'd use `function Navbar() { … }` (no leading space for parentheses) and four spaces for indentation (in Markdown anyway), plus `let`, semicolons and double quotes - but see maintainership note above.

Reply:
Yes, that is true.

I tend to use Navbar because I had a very bad experience in a previous product where I had to modify the navbar structure and it was _painful_ to update all of the dependent projects via Copy&Paste. We put it off for at least a few weeks.

Reply:
That's a good point: Perhaps we could start with a straightforward, self-explanatory example that doesn't rely on context/interpretation (not actually sure whether my `Card` qualifies for that tbh), but add afterwards that [the same principle applies to larger structures as well](https://en.wikipedia.org/wiki/Fractal) - a nav bar seems like a good example there; people should be able to grok that without the need for an elaborate code sample.

Reply:
I'm thinking about a new example that would word well.

By the way, I don't really like your card example because it always uses an `h3` as a card level, but the card would only know the correct semantic level in context of the document. 😬

For structural macros like cards I often use no parameters.

Reply:
I dislike the card example for the same reason. \o/ (Which is why I cheated above by omitting the heading... 😬 ) You're far better suited than me to come up with grokkable examples, but it might be worth considering using a simple [Bootstrap component](https://getbootstrap.com/docs/4.1/components) (list group, perhaps?) because they're widely known.